pyro/distributions/transforms/spline.py

```diff
@@ -33,15 +37,28 @@ def _select_bins(x, idx):
     Performs gather to select the bin in the correct way on batched inputs
     """
     idx = idx.clamp(min=0, max=x.size(-1) - 1)
-    extra_dims = len(idx.shape) - len(x.shape)
-    expanded_x = x.expand(idx.shape[:extra_dims] + (-1,) * len(x.shape))
-    return expanded_x.gather(-1, idx).squeeze(-1)
+
+    """
+    idx ~ (context_batch_dims, input_dim, 1)
+    x ~ (batch_dims, input_dim, count_bins)
+    """
+
+    # Broadcast dimensions of idx over x
```

Comment:
By the way, the reason I do this is because `gather` doesn't broadcast... Yes there must be a better way to ensure the input batch dims and the context variable batch dims line up

Reply:
I don't know a better way offhand.